src/configr_assign.hpp

```diff
@@ -26,7 +26,7 @@ struct configr_of
 		this->operator()(attrname, value);
 	}
 
-	configr_of(const config& cfg) : subtags_(), data_()
+	explicit configr_of(const config& cfg) : subtags_(), data_()
```

Comment:
Drop this commit.

Reply:
Sure. This is a bad code smell, anyway. The configr_of class seems stupid and probably should be a member function of config.

Reply:
Considering `config_of` was dropped, I can't seem to figure out what `configr_of` is good for.

Reply:
unlike config_of, configr_of  is not meant to construct config objects.

the intenetion was to allow wesnothd_connection to allow sending of nested config ojects to be sended without copying, for example here https://github.com/wesnoth/wesnoth/blob/1.13.11/src/game_initialization/connect_engine.cpp#L792 we copy the whole scenariowml,  just to temporarily wrap it into a [store_next_scenario]. It is not used yet though.

Reply:
@gfgtdf what's not used yet?

Reply:
I never really understood the point of `configr_of` either... I'd be in favour of dropping it

> we copy the whole scenariowml, just to temporarily wrap it into a [store_next_scenario]

Maybe this is solvable via something like a move constructor? Though I guess the scenario WML might be a read-only config so a move constructor won't work...

Reply:
The common use is something like `configr_of(config("emptytagcommand"))` to send that command to the server. I would have it become `config("emptytagcommand").to_string()` and the output would be the same, such as: `[emptytagcommand]\n[/emptytagcommand]\n`. That's basically what the `configr_of` class does, but faster and with less memory wasted.

The question is: is there something we're missing and does that prevent simply using a member function?

Whatever we decide, it's an issue for another PR. Here, I'm just concerned about clearing off as many 'missing explicit keyword' issues as possible so we can see when one which is actually important appears.